src/next/str.rs

```diff
@@ -71,6 +72,19 @@ impl core::str::FromStr for AccountId {
     }
 }
 
+impl core::fmt::Display for PoolId {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl core::str::FromStr for PoolId {
+    type Err = Error;
+    fn from_str(s: &str) -> core::result::Result<Self, Self::Err> {
+        Ok(PoolId(Hash::from_str(s)?))
+    }
+}
```

Comment:
These needs to use the strkey library change here:
- https://github.com/stellar/rs-stellar-strkey/pull/74

I was waiting on a review of the SEP-23 change here by @stanford-scs:
- https://github.com/stellar/stellar-protocol/pull/1646

Reply:
Yeah, I've opened an issue https://github.com/stellar/rs-stellar-xdr/issues/420 and added a TODO.